internal/envtest/envtest.go

```diff
@@ -3,69 +3,87 @@ package envtest
 import (
 	"context"
 	"path/filepath"
-	"runtime"
 	"strings"
 	"testing"
 
 	cassdcapi "github.com/k8ssandra/cass-operator/apis/cassandra/v1beta1"
 	controlapi "github.com/k8ssandra/cass-operator/apis/control/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
 	k8ssandrataskapi "github.com/k8ssandra/k8ssandra-operator/apis/control/v1alpha1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
 
-	"github.com/k8ssandra/k8ssandra-client/pkg/kubernetes"
-	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	"k8s.io/kubectl/pkg/scheme"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
 func Run(m *testing.M, setupFunc func(e *Environment)) (code int) {
-	env := NewEnvironment()
-	env.start()
+	ctx := ctrl.SetupSignalHandler()
+	env := NewEnvironment(ctx)
+	env.Start()
 	setupFunc(env)
 	exitCode := m.Run()
-	env.stop()
+	env.Stop()
 	return exitCode
 }
 
 type Environment struct {
-	intClient     client.Client
+	Client        client.Client
 	env           *envtest.Environment
 	cancelManager context.CancelFunc
 	Context       context.Context
+	Kubeconfig    string
 }
 
-func NewEnvironment() *Environment {
+func NewEnvironment(ctx context.Context) *Environment {
 	env := &Environment{}
 	env.env = &envtest.Environment{
-		CRDDirectoryPaths:     []string{filepath.Join(RootDir(), "testfiles", "crd")},
+		CRDDirectoryPaths: []string{
+			filepath.Join(RootDir(), "testfiles", "crd"),
+		},
 		ErrorIfCRDPathMissing: true,
 	}
-
-	ctx := ctrl.SetupSignalHandler()
 	ctx, cancel := context.WithCancel(ctx)
 	env.Context = ctx
 	env.cancelManager = cancel
 	return env
 }
 
-// https://stackoverflow.com/questions/31873396/is-it-possible-to-get-the-current-root-of-package-structure-as-a-string-in-golan
-func RootDir() string {
-	_, b, _, _ := runtime.Caller(0)
-	return filepath.Join(filepath.Dir(b), "../..")
+func (e *Environment) GetClient(namespace string) client.Client {
+	c, err := kubernetes.GetClientInNamespace(e.env.Config, namespace)
+	if err != nil {
+		panic(err)
+	}
+	return c
 }
 
-func (e *Environment) Client(namespace string) client.Client {
-	return client.NewNamespacedClient(e.intClient, namespace)
+func (e *Environment) RestConfig() *rest.Config {
+	return e.env.Config
+}
+
+func (e *Environment) RawClient() client.Client {
+	return e.Client
 }
 
-func (e *Environment) start() {
+func (e *Environment) Start() {
```

Comment:
question: Is there a reason to export Start() ? I can't see it being called outside this package.

Reply:
In my view, the two most important methods in this struct are Start and Stop. They should definitely be public. Again, I think this makes more sense in light of the next PR, since that separates the starting and stopping of the envtest environment from the backend.